disco_aws_automation/disco_elasticsearch_archive.py

```diff
@@ -0,0 +1,498 @@
+from ConfigParser import NoOptionError
+from collections import defaultdict
+from time import time, sleep
+import datetime
+import json
+import logging
+import re
+import os
+
+import boto3
+from boto3.session import Session
+from elasticsearch import (
+    Elasticsearch,
+    NotFoundError,
+    RequestsHttpConnection,
+    TransportError,
+)
+from requests_aws4auth import AWS4Auth
+
+from . import read_config, DiscoElasticsearch, DiscoIAM
+from .exceptions import TimeoutError
+from .disco_constants import (
+    ES_ARCHIVE_POLICIES_DIR,
+    ES_CONFIG_FILE
+)
+
+ES_SNAPSHOT_FINAL_STATES = ['SUCCESS', 'FAILED']
+SNAPSHOT_WAIT_TIMEOUT = 60*60
+SNAPSHOT_POLL_INTERVAL = 60
+ES_ARCHIVE_ROLE = "es_archive"
+
+
+class DiscoESArchive(object):
+    def __init__(self, environment_name, cluster_name, config_aws=None,
+                 config_es=None,
+                 index_prefix_pattern=None, repository_name=None, disco_es=None,
+                 disco_iam=None):
+        self.config_aws = config_aws or read_config()
+        self.config_es = config_es or read_config(ES_CONFIG_FILE)
+
+        if environment_name:
+            self.environment_name = environment_name.lower()
+        else:
+            self.environment_name = self.config_aws.get("disco_aws", "default_environment")
+
+        self.cluster_name = cluster_name
+        self._host = None
+        self._es_client = None
+        self._region = None
+        self._s3_client = None
+        self._s3_bucket_name = None
+        self._disco_es = disco_es
+        self._disco_iam = disco_iam
+        self.index_prefix_pattern = index_prefix_pattern or r'.*'
+        self.repository_name = repository_name or 's3'
+
+    @property
+    def host(self):
+        """
+        Return ES cluster hostname
+        """
+        if not self._host:
+            es_domains = self.disco_es.list()
+            try:
+                host = [es_domain["route_53_endpoint"]
+                        for es_domain in es_domains
+                        if es_domain.get("internal_name") == self.cluster_name][0]
+            except IndexError:
+                logging.info("Unable to find")
```

Comment:
logging statement can be improved .. or removed, as exception provides the details